src/graphics/GraphvizWiringDiagrams.jl

```diff
@@ -623,4 +624,64 @@ inverse_rank_dir(rank_dir::String) = @match rank_dir begin
   "RL" => RightToLeft
 end
 
+function to_graphviz(cp::CPortGraph; kw...)::Graphviz.Graph
+  to_graphviz(property_graph(cp; kw...))
+end
+
+""" Draw a circular port graph using Graphviz.
+
+Creates a Graphviz graph. Ports are currently not respected in the image, but
+the port index for each box can be displayed to provide clarification.
+
+# Arguments
+- `graph_name="G"`: name of Graphviz graph
+- `prog="neato"`: Graphviz program, usually "neato" or "fdp"
+- `box_labels=false`: whether to label boxes with their number
+- `port_labels=false`: whether to label ports with their number
+- `graph_attrs=Dict()`: top-level graph attributes
+- `node_attrs=Dict()`: top-level node attributes
+- `edge_attrs=Dict()`: top-level edge attributes
+
+# TODO:
+The lack of ports might be able to be resolved by introducing an extra node per
+port which is connected to its box with an edge of length 0.
+"""
+
+function property_graph(cp::CPortGraph;
+    graph_name::String="G", prog::String="neato",
+    box_labels::Bool=false, port_labels::Bool=false,
+    graph_attrs::AbstractDict=Dict(), node_attrs::AbstractDict=Dict(),
+    edge_attrs::AbstractDict=Dict())
+  default_attrs = default_undirected_graphviz_attrs
+
+  # Generate a map from the global index of a port to the box-specific index
+  port2box = Array{Tuple{Int, Int}}(undef, nparts(cp, :Port))
+  for (b, p) in enumerate(incident(cp, 1:nparts(cp, :Box), :box))
+    port2box[p] .= [(b, i) for i in 1:length(p)]
+  end
+
+  g′ = Graphs.Graph()
+
+  add_parts!(g′, :V, nparts(cp, :Box))
+  add_parts!(g′, :E, nparts(cp, :Wire), src=cp[cp[:src], :box], tgt=cp[cp[:tgt], :box])
```

Comment:
This looks like a pullback data migration with `V => Box`, `E => Wire`, `src => compose(src, box)` and `tgt => compose(tgt, box)`.

Reply:
Didn't realize you could do composition with the data migration tools, that's neat! I've changed the implementation to using that now.